src/cmd_parser.c

```diff
@@ -146,6 +146,14 @@ static GOptionEntry cmd_entries[] =
       "Number of workers to spawn to read rpms.", NULL },
     { "xz", 0, 0, G_OPTION_ARG_NONE, &(_cmd_options.xz_compression),
       "Use xz for repodata compression.", NULL },
+    { "zck", 0, 0, G_OPTION_ARG_NONE, &(_cmd_options.zck_compression),
+      "Generate zchunk files as well as the standard repodata.", NULL },
+    { "zck-primary-dict", 0, 0, G_OPTION_ARG_FILENAME, &(_cmd_options.zck_primary_dict),
+      "Compression dictionary to use for zchunk primary file", "ZCK_PRIMARY_DICT" },
+    { "zck-filelists-dict", 0, 0, G_OPTION_ARG_FILENAME, &(_cmd_options.zck_filelists_dict),
+      "Compression dictionary to use for zchunk filelists file", "ZCK_FILELISTS_DICT" },
+    { "zck-other-dict", 0, 0, G_OPTION_ARG_FILENAME, &(_cmd_options.zck_other_dict),
+      "Compression dictionary to use for zchunk other file", "ZCK_OTHER_DICT" },
```

Comment:
Are the dictionaries mandatory, or optional? My understand of zstd is that they make it better for compression, but aren't necessary.

Reply:
They are optional, but make a significant difference in the size of the zchunked metadata.  I've put together dictionaries for primary, filelists and other based on data from current Fedora releases, that I'd suggest packaging for Fedora/EPEL (if that's what our builders are on).

The dictionary is embedded in the zchunk file, so clients don't need a copy.

If the dictionary changes, there will be no common chunks between the pre-change and post-change files, so dictionary changes should happen a maximum of once a release (and probably aren't needed even that often)

Reply:
Forgive me for being forgetful, but is there a script or something that people can use to generate dictionaries for their distributions' metadata?

Reply:
I've got a [small python script](https://www.jdieter.net/downloads/zchunk-dicts/split.py) that splits the metadata into much smaller files and strips out any checksums (which by definition are unpredictable).

I ran the script over a couple of days worth of updates metadata, which produced thousands of files, and then, in the directory containing those files, ran:
```
zstd --train --dictID=0 *
```

The default dict size of roughly 100KB was fairly optimal for Fedora's metadata.  Too small and you don't see the savings you'd expect.  Too large and you're just wasting space.

Reply:
The proposed Fedora dicts are available at [https://www.jdieter.net/downloads/zchunk-dicts](https://www.jdieter.net/downloads/zchunk-dicts)

Reply:
I've gone ahead and updated the man page.  I'll work on the tests but it might take a bit longer to get them done.